### PR TITLE
Fix pre-1880 date selection

### PR DESCRIPTION
**Pull request by phtucker**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Day info objects for Calendar are being populated with inconsistent dates due to an issue with converting pre-1880 dates to and from strings. This change populates day info objects from a single date instance preventing the previous inconsistency.

#### Focus areas to test

DatePicker

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/9857)

### CODE REVIEW

**Comment by Vitalius1** [comment]
@phtucker thanks for submitting this PR. `Calendar` in OUFR will be replaced by the one in `date-time` package. I see that this issue is present there too. Can you please update it in both places:
https://github.com/OfficeDev/office-ui-fabric-react/blob/6340860825064e5f12e6f569cc2b7ee6859b7a8e/packages/date-time/src/components/CalendarDayGrid/CalendarDayGrid.base.tsx#L349-L358

Also, will wait on @lorejoh12 to approve as he is most familiar with the `Calendar` logic.

**Comment by lorejoh12** [comment]
Please do include the fix for the date-time control also.

I dug a little bit, and this is a really interesting bug. Turns out, the reason this bug exists is because the date Nov 18, 1883 is missing from our calendar.
image

This was the day that Railroad time was implemented in the United States, so the beginning of time zones. Guess this is probably a weird time-zone conversion bug going on.

Here's what happens with your fix:
image

Basically, the time goes from selecting at midnight on the wrong day to selecting at 11:50pm on the correct day. So at least the date is correct (it's 11:52pm on the wrong day before your fix). It doesn't actually add back the mysteriously missing day though. Also, the entire month of September is blank in your build, but not in the official build. Not entirely sure why.

**Comment by phtucker** [comment]
Interesting, I'll have a look at the September issue, thanks.

**Comment by Vitalius1** [comment]
@phtucker can you please look into the September issue @lorejoh12 mentioned and push an update so that we can look into merging the fix. It would also be recommended to pull in the current master to make sure everything still builds and there are no conflicts. Keep in mind that we migrated our build tools into using Lerna and Yarn so the commands slightly changed. You can find [info in the wiki](https://github.com/OfficeDev/office-ui-fabric-react/wiki/Build-Commands) about it.

Thanks!

**Comment by phtucker** [comment]
I will probably be abandoning this PR as the issue it causes is worse than the issue that it fixes, and less clear as to why it is occurring.

**Comment by Vitalius1** [comment]
I will close the PR then and thanks for the attempt. We welcome you to update this one and reopen it or come with a separate one if you still going to be looking into the issue.

@lorejoh12 FYI
